delta-lake/src/test/java/org/apache/iceberg/delta/TestDeltaLakeTypeToType.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.delta;
+
+import io.delta.standalone.types.ArrayType;
+import io.delta.standalone.types.BinaryType;
+import io.delta.standalone.types.BooleanType;
+import io.delta.standalone.types.DoubleType;
+import io.delta.standalone.types.LongType;
+import io.delta.standalone.types.MapType;
+import io.delta.standalone.types.StringType;
+import io.delta.standalone.types.StructType;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestDeltaLakeTypeToType {
+  private static final String optionalBooleanType = "testNullableBoolType";
+
+  private static final String requiredBinaryType = "testRequiredBinaryType";
+
+  private static final String doubleArrayType = "testNullableArrayType";
+
+  private static final String innerAtomicSchema = "testInnerAtomicSchema";
+
+  private static final String stringLongMapType = "testStringLongMap";
+
+  private StructType deltaAtomicSchema;
+
+  private StructType deltaNestedSchema;
+
+  @Before
+  public void constructDeltaLakeSchema() {
+    deltaAtomicSchema =
+        new StructType()
+            .add(optionalBooleanType, new BooleanType())
+            .add(requiredBinaryType, new BinaryType(), false);
+    deltaNestedSchema =
+        new StructType()
+            .add(innerAtomicSchema, deltaAtomicSchema)
+            .add(doubleArrayType, new ArrayType(new DoubleType(), true), false)
+            .add(stringLongMapType, new MapType(new StringType(), new LongType(), false), false);
+  }
+
+  @Test
+  public void testAtomicTypeConversion() {
+    Type converted =
+        DeltaLakeDataTypeVisitor.visit(
+            deltaAtomicSchema, new DeltaLakeTypeToType(deltaAtomicSchema));
+    Schema convertedSchema = new Schema(converted.asNestedType().asStructType().fields());
+    Assert.assertTrue(
+        "The BooleanType should be converted to BooleanType",
+        convertedSchema.findType(optionalBooleanType) instanceof Types.BooleanType);
```

Comment:
when this test fails then the output will be `true != ..`, therefore it would be better to change this to `Assertions.assertThat(convertedSchema.findType(optionalBooleanType)).isInstanceOf(Types.BooleanType.class);`. That will show the class of the returned type in case the assertion ever fails.

That being said, I'd probably change these 4 assertions to
```
Assertions.assertThat(convertedSchema.findType(optionalBooleanType)).isInstanceOf(Types.BooleanType.class);
Assertions.assertThat(convertedSchema.findField(optionalBooleanType).isOptional()).isTrue();
Assertions.assertThat(convertedSchema.findType(requiredBinaryType)).isInstanceOf(Types.BooleanType.class);
Assertions.assertThat(convertedSchema.findField(requiredBinaryType).isRequired()).isTrue();
```

Same for all the other methods